packages/code-du-travail-frontend/pages/newsletter.js

```diff
@@ -0,0 +1,243 @@
+
+const STATUS = {
+  INITIAL: "initial",
+  INVALID: "invalid",
+  LOADING: "loading",
+  REGISTERED: "registered",
+  ERRORED: "errored"
+};
+
+const Newsletter = ({ pageUrl, ogImage }) => {
+  const [email, setEmail] = useState("");
+  const [status, setStatus] = useState(STATUS.INITIAL);
+
+  const checkEmail = useCallback(email => {
+    if (/.+@.+\..+/.test(email)) {
+      setStatus(STATUS.INITIAL);
+      return true;
+    }
+    setStatus(STATUS.INVALID);
+    return false;
+  }, []);
+
+  const onSubmit = useCallback(
+    async email => {
+      if (!checkEmail(email, setStatus)) {
+        return;
+      }
+      try {
+        setStatus(STATUS.LOADING);
+        const response = await fetch(
+          `https://sarbacaneapis.com/v1/lists/${SARBACANE_LIST_ID}/contacts/upsert`,
+          {
+            method: "POST",
+            headers: new Headers({
+              accountId: SARBACANE_ACCOUNT_ID
+            }),
+            body: JSON.stringify({ email })
+          }
+        );
+        if (response.ok) {
+          setStatus(STATUS.REGISTERED);
+        } else {
+          setStatus(STATUS.ERRORED);
+          const data = await response.json();
+          Sentry.captureException(data.message);
+        }
+      } catch (error) {
+        setStatus(STATUS.ERRORED);
+        Sentry.captureException(error);
+      }
+    },
+    [checkEmail]
+  );
+
+  let currentView = null;
+  switch (status) {
+    case STATUS.REGISTERED:
+      currentView = (
+        <CenterDiv>
+          <Toast shadow animate="from-top">
+            <StyledHeading as={Heading}>
+              L’enregistrement s’est bien effectué.
+            </StyledHeading>
+            <CenterDiv>
+              Merci de l’intérêt que vous portez au Code du Travail
+              numérique&nbsp;!
+              <br />
+              <Link href="/" passHref>
+                <Button variant="primary" as={StyledLink}>
+                  Retourner sur la page d’accueil <StyledArrowRight />
+                </Button>
+              </Link>
+            </CenterDiv>
+          </Toast>
+        </CenterDiv>
+      );
+      break;
+    case STATUS.ERRORED:
+      currentView = (
+        <CenterDiv>
+          <Toast shadow variant="primary" animate="from-top">
+            <StyledHeading as={Heading}>
+              Oups, une erreur est survenue pendant l’inscription.
+            </StyledHeading>
+            <CenterDiv>
+              Nos équipes ont été informées.
+              <br />
+              <Link href="/" passHref>
+                <Button variant="primary" as={StyledLink}>
+                  Retourner sur la page d’accueil <StyledArrowRight />
+                </Button>
+              </Link>
+            </CenterDiv>
+          </Toast>
+        </CenterDiv>
+      );
+      break;
+    default:
+      currentView = (
+        <Wrapper variant="main">
+          <Form
+            onSubmit={e => {
+              e.preventDefault();
```

Comment:
this should be done in useCallback so you don't create a new function on each render

Reply:
Bonne question ! Je crois qu'au fil de refacto j'ai pas pensé à mettre le preventDefault dans le onSubmit !